### PR TITLE
Nblib add docs to topology & sim state

### PR DESCRIPTION
**Pull request by kanduri**
Just began writing docs for `SimulationState` and `TopologyBuilder`. Will write more for other classes as well as the API finalizes.

### CODE REVIEW

File: src/gromacs/nblib/simulationstate.h [hunk 1]
```diff
@@ -58,34 +58,44 @@
...
-//! Simulation state description that serves as a snapshot of the system
-//! being analysed. Needed to init an MD program. Allows hot-starting simulations.
+/*! \libinternal
+ * \ingroup nblib
+ * \brief Simulation State
+ *
+ * Simulation state description that serves as a snapshot of the system
+ * being analysed. Needed to init an MD program. Allows hot-starting simulations.
+ */
+
```
Comment (ejjordan) [inline src/gromacs/nblib/simulationstate.h]: I think we don't want spaces between the doxygen and the class it documents.

File: src/gromacs/nblib/topology.h [hunk 2]
```diff
@@ -94,22 +94,45 @@ class Topology {
...
 
+/*! \libinternal
+ * \ingroup nblib
+ * \brief Topology Builder
+ *
+ * A helper class to assist building of topologies. They also ensure that
+ * topologies only exist in a valid state within the scope of the
+ * simulation program.
+ */
+
```
Comment (ejjordan) [inline src/gromacs/nblib/topology.h]: same

**Review by ejjordan — CHANGES_REQUESTED** [review]
LGTM after a trivial change.

File: src/gromacs/nblib/simulationstate.h (hunk 1 above)
Comment (kanduri) [reply src/gromacs/nblib/simulationstate.h]: Done

Comment (ejjordan) [reply src/gromacs/nblib/simulationstate.h]: It was still the same but I uploaded a small fix.

**Review by ejjordan — APPROVED** [review]

**Pull request merged**